### PR TITLE
Develop feature overwrite

### PR DESCRIPTION
**Pull request by jnz86**
Tilen, 

Like I said, it wouldn't be super pretty. There may even be an edge case I didn't consider. But I didn't have a ton of time to make this and have other work to do. 

Take a look and see if there is anything you want from it!

### CODE REVIEW

**Comment by jnz86** [comment]
Added copy().

I did what I could in the time I had!

File: lwrb/src/lwrb/lwrb_ex.c [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+ * \brief           Copy one ring buffer to another, upto the amount of data in the source, or amount 
+ * of data free in the destination.
+ * \param[in]       dest: Buffer handle that the copied data will be written to
+ * \param[in]       src:  Buffer handle that the copied data will come from
+ * \return          Number of bytes written to buffer
+ * \note            This operation is a read op to the source, on success it will update the r index. 
+ * As well as a write op to the destination, and may update the w index. For thread-safety mutexes may 
+ * be desired, see documentation.
+ */
+size_t lwrb_copy(lwrb_t* dest, lwrb_t* src)
```
Comment (MaJerle) [inline lwrb/src/lwrb/lwrb_ex.c:120]: This function could be implemented using `lwrb_get_linear_read_length` and same for write. You get direct length of data that can be copied 1-to-1 with one memcpy call.

**Comment by MaJerle** [comment]
Thanks - I will merge on new branch, run formatting and add some code rework for optimizations.

File: lwrb/src/lwrb/lwrb_ex.c (hunk 1 above)
Comment (MaJerle) [reply lwrb/src/lwrb/lwrb_ex.c:120]: You also call the free and full before checking if input buffer is valid. I believe a src lwrb should also be const, etc.

I will work on this during the weekend

**Pull request merged**